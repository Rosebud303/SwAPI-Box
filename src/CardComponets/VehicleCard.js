import React from 'react';
import './VehicleCard.scss'

export default function VehicleCard({vehicle}) {
  return (
    <div className='vehicleCard'>
        <h3 className='vehicleName'>{vehicle.Name}</h3>
        <p>Model: {vehicle.Model}</p>
        <p>Class: {vehicle.Class}</p>
        <p>Passengers: {vehicle.Passengers}</p>
        <button>Add Favorite</button>
    </div>
  )
}
