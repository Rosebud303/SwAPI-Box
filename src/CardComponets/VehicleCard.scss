.vehicleCard {
    width: 320px;
    text-align: center;
    margin-bottom: 20px;
    margin-right: 20px;
    border: 4px solid #F44336;
    border-radius: 25px;
    padding-bottom: 10px;
    background: #FFC107;
    box-shadow: 4px 4px 4px #968888;
}

.vehicleName {
    background: rgb(220, 182, 57);
    height: 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 20px;
}