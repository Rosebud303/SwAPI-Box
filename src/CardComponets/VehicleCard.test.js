import React from 'react';
import VehicleCard from './VehicleCard';
import { shallow } from 'enzyme';

describe('VehicleCard', () => {
    let wrapper;
    let mockVehicle = {name: 'sandCrawler', passengers: 30}

    beforeEach(() => {
        wrapper = shallow(
            <VehicleCard vehicle={mockVehicle}
                        key={mockVehicle.name}
            />
        )
    })

    it('should match the snap shot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})