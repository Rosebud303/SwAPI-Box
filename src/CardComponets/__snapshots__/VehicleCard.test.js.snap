// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match the snap shot 1`] = `ShallowWrapper {}`;
